etna/datasets/tsdataset.py

```diff
@@ -815,7 +815,7 @@ def to_hierarchical_dataset(
         """
         df_copy = df.copy(deep=True)
         df_copy["segment"] = df_copy[level_columns].astype("string").agg(sep.join, axis=1)
-        if not keep_level_columns:
+        if not keep_level_columns and len(level_columns) > 0:
```

Comment:
I'm not really sure if we should do this check with `len` here.

Reply:
In fact we don't as this method expect to exist at least one column in `level_columns`

Reply:
May be we can throw error in the beginning if it is empty